pkg/controller/kogitoinfra/mongodb.go

```diff
@@ -0,0 +1,308 @@
+		envVarMongoDBDatabase:     "QUARKUS_MONGODB_DATABASE",
+	}
+	// propertiesMongoDBSpring mongoDB properties for spring boot runtime
+	propertiesMongoDBSpring = map[int]string{
+		appPropMongoDBHost: "spring.data.mongodb.host",
+		appPropMongoDBPort: "spring.data.mongodb.port",
+
+		envVarMongoDBAuthDatabase: "SPRING_DATA_MONGODB_AUTHENTICATION_DATABASE",
+		envVarMongoDBUser:         "SPRING_DATA_MONGODB_USERNAME",
+		envVarMongoDBPassword:     "SPRING_DATA_MONGODB_PASSWORD",
+		envVarMongoDBDatabase:     "SPRING_DATA_MONGODB_DATABASE",
+	}
+)
+
+func getMongoDBSecretEnvVars(cli *client.Client, mongoDBInstance *mongodb.MongoDB, instance *v1alpha1.KogitoInfra, scheme *runtime.Scheme) ([]corev1.EnvVar, error) {
+	var envProps []corev1.EnvVar
+
+	customMongoDBSecret, resultErr := loadCustomKogitoMongoDBSecret(cli, instance.Namespace)
+	if resultErr != nil {
+		return nil, resultErr
+	}
+
+	if customMongoDBSecret == nil {
+		customMongoDBSecret, resultErr = createCustomKogitoMongoDBSecret(cli, instance.Namespace, mongoDBInstance, instance, scheme)
+		if resultErr != nil {
+			return nil, resultErr
+		}
+	}
+
+	envProps = append(envProps, framework.CreateEnvVar(mongoDBEnablePersistenceEnvKey, "true"))
+	mongoDBSecretName := customMongoDBSecret.Name
+	envProps = append(envProps, framework.CreateEnvVar(mongoDBEnvKeyCredSecret, mongoDBSecretName))
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBSpring[envVarMongoDBAuthDatabase], mongoDBSecretName, infrastructure.MongoDBAppSecretAuthDatabaseKey))
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBQuarkus[envVarMongoDBAuthDatabase], mongoDBSecretName, infrastructure.MongoDBAppSecretAuthDatabaseKey))
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBSpring[envVarMongoDBUser], mongoDBSecretName, infrastructure.MongoDBAppSecretUsernameKey))
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBQuarkus[envVarMongoDBUser], mongoDBSecretName, infrastructure.MongoDBAppSecretUsernameKey))
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBSpring[envVarMongoDBPassword], mongoDBSecretName, infrastructure.MongoDBAppSecretPasswordKey))
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBQuarkus[envVarMongoDBPassword], mongoDBSecretName, infrastructure.MongoDBAppSecretPasswordKey))
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBSpring[envVarMongoDBDatabase], mongoDBSecretName, infrastructure.MongoDBAppSecretDatabaseKey))
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBQuarkus[envVarMongoDBDatabase], mongoDBSecretName, infrastructure.MongoDBAppSecretDatabaseKey))
+	return envProps, nil
+}
+
+func getMongoDBAppProps(cli *client.Client, mongoDBInstance *mongodb.MongoDB) (map[string]string, error) {
+	appProps := map[string]string{}
+
+	mongoDBURI := mongoDBInstance.Status.MongoURI
```

Comment:
Why not using the attribute directly and drop this temp?

Reply:
I just don't want to repeat it 3 times into the code